### DISCUSSION TITLE
Stumpy for Clustering and later Classification

### DISCUSSION THREAD

**Discussion by zyberzebra** [TDAmeritrade/stumpy#404] [Q&A]
I thought using stumpy for clustering and so i set up an example.

# What i did

I have 3 Patterns i concatenated multiple times and generated a time series.

![image](https://user-images.githubusercontent.com/73640367/121689319-c2785980-cac4-11eb-9fa6-f0893c8e1d19.png)

I know there length is 150.

---

# Questions

How do i make stumpy find those 3 patterns ?
Shouldnt that be an easy task for stumpy? 
There is no noice and they are exactly the same every time, so i guess it should work with my example
Is there a way i can see how many clusters where found, and how many of each where found ?

# Code
```python
import stumpy
m = 150
mp = stumpy.stump(time_series,m=m)
```
Result:
![image](https://user-images.githubusercontent.com/73640367/121690116-b93bbc80-cac5-11eb-9ec0-0fe9b07a5895.png)

**Comment by seanlaw** [comment]
@marius-mm Welcome to the STUMPY community. Can you please share your code for what is being plotted? Additionally, can you show what three patterns you are referring to?

Perhaps, there is a misunderstanding but STUMPY is not a time series clustering tool. Instead, the matrix profile that is computed gives you the (one) nearest neighbor information to each subsequence.

**Reply by zyberzebra** [reply]
ahahah guess i started implementing to fast before reading the introduction.
could it be used for early classification?

**Reply by seanlaw** [reply]
This is a "work-in-progress" tutorial may be of interest to you as it relates to building a classification model:

https://github.com/TDAmeritrade/stumpy/blob/main/docs/Tutorial_Shapelet_Discovery.ipynb

**Reply by JaKasb** [reply]
* If you have labelled data, you can use shapelets.
* If your subsequences are non-overlapping (whole-sequence clustering), you can use any clustering method by using the pairwise distances, located at every m-th index of the MatrixProfile. 
* If you want to perform unsupervised sliding-window clustering, you need a research group and 2 years. 

If you are interested in the research on sliding-window subsequence-clustering, I can provide some papers.
However I am not aware of a simple and established method.

**Reply by zyberzebra** [reply]
> * If you want to perform unsupervised sliding-window clustering, you need a research group and 2 years.

Damn.. Guess i need to rethink about what im trying to do...
